modules/plugin/jdbc/jdbc-postgis/src/main/java/org/geotools/data/postgis/PostgisNGDataStoreFactory.java

```diff
@@ -85,6 +85,12 @@
     public static final Param SIMPLIFY = new Param("Support on the fly geometry simplification", Boolean.class, 
             "When enabled, operations such as map rendering will pass a hint that will enable the usage of ST_Simplify", false, Boolean.TRUE);
     
+    /**
+     * Enables usage of ST_Envelope to wrap geometries on Overlap filters, to speed up queries (GEOT-5167)
+     */
+    public static final Param ENCODE_BBOX_FILTER_USING_ENVELOPE = new Param("Wrap geometries with Envelope function in Overlap filters", Boolean.class, 
+            "When enabled, ST_Envelope is wrapper around geometries in overlap queries, to speed-up some edge use cases", false, Boolean.FALSE);
+    
```

Comment:
The "some edge use cases" is very developer focused, this description is used in user interfaces shown to real. people. The "key" used in a parameter map is verbose for a map key ...

* KEY: Wrap geometries with Envelope function in Overlap filters <-- more a 
* DESCRIPTION: "When enabled, ST_Envelope is wrapper around geometries in overlap queries, to speed-up some edge use cases"

Propose:

* key: "overlap_optimisation"
* description: "Enables use of ST_Envelope to wrap geometries on Overlap filter"